docs/architecture/adrs/007-orchestrator-telemetry.md

```diff
@@ -0,0 +1,74 @@
+for OpenTelemetry in Rust, and exporting it through the OTLP exporter if configured. The OTLP exporter will send 
+telemetry data to a gRPC or HTTP endpoint that can be configured to point to a running OTEL collector.
+Similarly, detectors should also be able to collect and export telemetry through OTLP to the same OTEL collector.
+From the OTEL collector, the telemetry data can then be exported to multiple backends. The OTEL collector and
+any observability backends can all be configured alongside the orchestrator and detectors in a deployment.
+
+### Instrumentation
+An incoming request to the orchestrator will initialize a new trace, therefore a trace-id and request should be in
+one-to-one correspondence. All important functions throughout the control flow of handling a request in the orchestrator 
+will be instrumented with the `#[tracing::instrument]` attribute macro. This will create and enter a span for each 
+function call and add it to the trace of the request. Here, important functions refers to any function that performs
+important business logic or that may incur significant latency, this includes all the handler functions for incoming
+and outgoing requests.
+
+### Metrics
+The orchestrator will aggregate metrics regarding the requests it has received/handled, and annotate the metrics with
+span attributes allowing for detailed filtering and monitoring. The metrics will be exported through the OTLP exporter
+through the metrics provider. Traces exported through the traces provider can also have R.E.D. (request, error and
+duration) metrics attached to them implicitly by the OTEL collector using the `spanmetrics` connector. Both the OTLP
+metrics and the `spanmetrics` metrics can be exported to configured metrics backends like Prometheus or Grafana.
+The orchestrator will handle a variety of useful metrics such as counters and histograms for received/handled 
+successful/failed requests, request and stream durations, and server/client errors.
+
+### Configuration
+The orchestrator will expose CLI args/env variables for configuring the OTLP exporter endpoint(s) 
+(e.g. `OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4317`) and protocol(s) (e.g. `OTEL_EXPORTER_OTLP_PROTOCOL=grpc`), 
+as well as the ability to enable trace (includes logs) and metrics OTLP exporting (e.g. `OTLP_EXPORT=traces,metrics`.
+Alongside this, the orchestrator will also allow for configuring the Rust log level (e.g. `RUST_LOG=debug`) and 
+log format to use for logs printed to stdout (e.g. `STDOUT_LOG=pretty`). Printing logs to stdout can also be disabled
+(e.g. `STDOUT_LOG=none`). The OTEL collector can be configured to receive from the OTLP endpoint(s) and export to
+observability backends. Examples of a complete configuration for the orchestrator and OTEL collector can be found in
+the playbooks [LINK HERE] and [LINK HERE] as Docker Compose files and as an OpenShift deployment.
```

Comment:
nit: can we reformat this to points, so that its easier to collect what all environment variables are getting added with this ?